pkg/pubsub/adapter/converters/sheduler.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package converters
+
+import (
+	"context"
+	"errors"
+	cloudevents "github.com/cloudevents/sdk-go"
+	. "github.com/cloudevents/sdk-go/pkg/cloudevents"
+	cepubsub "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/pubsub"
+	pubsubcontext "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/pubsub/context"
+	"github.com/google/knative-gcp/pkg/apis/events/v1alpha1"
+)
+
+const (
+	SchedulerAdapterType = "com.google.cloud.scheduler"
+)
+
+func convertScheduler(ctx context.Context, msg *cepubsub.Message, sendMode ModeType) (*cloudevents.Event, error) {
+	tx := pubsubcontext.TransportContextFrom(ctx)
+	// Make a new event and convert the message payload.
+	event := cloudevents.NewEvent(cloudevents.VersionV1)
+	event.SetID(tx.ID)
+	event.SetTime(tx.PublishTime)
+	// We do not know the content type and we do not want to inspect the payload,
+	// thus we set this generic one.
+	event.SetDataContentType("application/octet-stream")
+	event.SetType(v1alpha1.SchedulerEventType)
```

Comment:
we should update the CRD yaml as well, to include this new information in the registry.... Without schema, but we can copy some description from the website.